### PR TITLE
[GLUTEN-6483] Support Uniffle 0.9.0

### PR DESCRIPTION
**Pull request by SteNicholas**
## What changes were proposed in this pull request?

Apache Uniffle has released v0.9.0, of which release note could refer to [v0.9.0](https://github.com/apache/incubator-uniffle/releases/tag/v0.9.0) for more details. This pull request is aimed to support v0.9.0 of Uniffle.

Fixes: #6483.

## How was this patch tested?

CI.

### CODE REVIEW

**Comment by SteNicholas** [comment]
Ping @summaryzb.

**Comment by zuston** [comment]
Try to attach `spark.rss.client.assignment.shuffle.nodes.max=1`  into the uniffle client config.

**Comment by SteNicholas** [comment]
@zuston, @summaryzb, I have attached `spark.rss.client.assignment.shuffle.nodes.max=1` in integration test. But the CI failed with `There isn't enough shuffle servers`. Should the version upgrade introduce extra configuration of uniffle?

**Comment by xianjingfeng** [comment]
> @zuston, @summaryzb, @xianjingfeng, I have attached `spark.rss.client.assignment.shuffle.nodes.max=1` in integration test. But the CI failed with `There isn't enough shuffle servers`. Should the version upgrade introduce extra configuration of uniffle via https://github.com/apache/incubator-uniffle/pull/1753?

You can try to set rss.rpc.server.type  to GRPC_NETTY

**Comment by SteNicholas** [comment]
@xianjingfeng, I have set `rss.rpc.server.type`  to `GRPC_NETTY` in `server.conf`. Unfortunately there is the same exception `There isn't enough shuffle servers`.

**Comment by xianjingfeng** [comment]
@SteNicholas 
https://github.com/apache/incubator-gluten/blob/0c519c4d5f7bfabc968e65d857b037f46058f51b/.github/workflows/velox_docker.yml#L507
https://github.com/apache/incubator-gluten/blob/0c519c4d5f7bfabc968e65d857b037f46058f51b/.github/workflows/velox_docker.yml#L519

**Comment by xianjingfeng** [comment]
Do you have a solution for the current CI failure? @SteNicholas

**Comment by summaryzb** [comment]
Maybe it's better to add mvn options in here https://github.com/apache/incubator-gluten/pull/6484/files#diff-38eda6989d285e6a3d79a3a66cc656d0b6d483b3d5c77bd0969d878642fd0f0eL512

**Comment by SteNicholas** [comment]
Ping @summaryzb, @xianjingfeng. PTAL.

**Review by xianjingfeng — APPROVED** [review]
LGTM

File: .github/workflows/velox_docker.yml [hunk 1]
```diff
@@ -498,32 +498,40 @@ jobs:
...
+          bash -c "echo -e 'rss.server.app.expired.withoutHeartbeat 7200000\nrss.server.heartbeat.delay 3000\nrss.rpc.server.port 19997\nrss.rpc.server.type GRPC_NETTY\nrss.jetty.http.port 19996\nrss.server.netty.port 19995\nrss.storage.basePath /opt/uniffle/shuffle_data\nrss.storage.type MEMORY_LOCALFILE\nrss.coordinator.quorum localhost:19999\nrss.server.flush.thread.alive 10\nrss.server.single.buffer.flush.threshold 64m' > ./conf/server.conf" && \
           bash ./bin/start-coordinator.sh && bash ./bin/start-shuffle-server.sh
-          cd $GITHUB_WORKSPACE/tools/gluten-it && $MVN_CMD clean install -Pspark-3.2 -Puniffle && \
+      - name: Build for Spark ${{ matrix.spark }}
+        run: |
+          cd $GITHUB_WORKSPACE/ && \
+          export MAVEN_HOME=/usr/lib/maven && \
+          export PATH=${PATH}:${MAVEN_HOME}/bin && \
+          $MVN_CMD clean install -P${{ matrix.spark }} -Pbackends-velox -Puniffle -DskipTests
```
Comment (jackylee-ch) [inline .github/workflows/velox_docker.yml:527]: We need remove the upper duplicated `Build for Spark`.

Comment (SteNicholas) [reply .github/workflows/velox_docker.yml:521]: @jackylee-ch, done.

File: .github/workflows/velox_docker.yml [hunk 2]
```diff
@@ -511,11 +511,11 @@ jobs:
           sed -i '226d' ./server/src/main/java/org/apache/uniffle/server/ShuffleTaskManager.java && \
```
[the remark this answers is not in the document]
Comment (SteNicholas) [reply .github/workflows/velox_docker.yml]: @summaryzb, this pull request has already directly used the release archive.

**Review by jackylee-ch — APPROVED** [review]
LGTM, thanks for your work.

File: .github/workflows/velox_docker.yml [hunk 3]
```diff
@@ -492,38 +492,39 @@ jobs:
...
+          bash -c "echo -e 'rss.server.app.expired.withoutHeartbeat 7200000\nrss.server.heartbeat.delay 3000\nrss.rpc.server.port 19997\nrss.rpc.server.type GRPC_NETTY\nrss.jetty.http.port 19996\nrss.server.netty.port 19995\nrss.storage.basePath /opt/uniffle/shuffle_data\nrss.storage.type MEMORY_LOCALFILE\nrss.coordinator.quorum localhost:19999\nrss.server.flush.thread.alive 10\nrss.server.single.buffer.flush.threshold 64m' > ./conf/server.conf" && \
           bash ./bin/start-coordinator.sh && bash ./bin/start-shuffle-server.sh
-          cd $GITHUB_WORKSPACE/tools/gluten-it && $MVN_CMD clean install -Pspark-3.2 -Puniffle && \
+      - name: Build for Spark ${{ matrix.spark }}
+        run: |
+          cd $GITHUB_WORKSPACE/ && \
+          export MAVEN_HOME=/usr/lib/maven && \
+          export PATH=${PATH}:${MAVEN_HOME}/bin && \
```
Comment (PHILO-HE) [inline .github/workflows/velox_docker.yml:519]: Looks the above two lines are redundant?

Comment (SteNicholas) [reply .github/workflows/velox_docker.yml:519]: @PHILO-HE, this just move the position of `Build for Spark`. Meanwhile, these lines are not redundant for `MVN_CMD`.

Comment (PHILO-HE) [reply .github/workflows/velox_docker.yml:518]: @SteNicholas, I see. Env variables exported in one step is not visible to the following step.

**Comment by SteNicholas** [comment]
@PHILO-HE, the CI of uniffle has already been successful. Please help to merge.

**Review by jackylee-ch — APPROVED** [review]

**Comment by jackylee-ch** [comment]
Just wait for the CI finished. After that, I would merge this pr if no more comments.

**Pull request merged**